### ISSUE TITLE
ContextFactory wrongly detect use statement

### ISSUE THREAD

**Issue by maMykola** [phpDocumentor/TypeResolver#191]
The `ContextFactory::createForNamespace()` wrongly detect use statements for class, f.e. if I have closure with `use` statement it's detected as a use statement for trait, but shouldn't:

```php
trait MyTrait
{
    protected function check(array $data): void
    {
        $key = 'test';

        array_walk($data, function (&$item) use ($key) {
            // modify $item
        });
    }
}
```

**Comment by maMykola** [comment]
Perhaps it'll be safe to add `case T_TRAIT:` alongside with `T_CLASS` https://github.com/phpDocumentor/TypeResolver/blob/1.x/src/Types/ContextFactory.php#L183

**Comment by maMykola** [comment]
Here is the fix for this issue https://github.com/phpDocumentor/TypeResolver/pull/192

**Comment by jaapio** [comment]
Thanks for your report, and fix this issue can be closed.

**Comment by jaapio** [comment]
I released 1.7.3 with your bugfix.
